### PR TITLE
Add `Integer#round` and `Float#round` core extensions

### PR DESCRIPTION
**Pull request by yui-knk**
`half` option was added to Integer#round and Float#round in Ruby 2.4.
And a behavior of round was changed to round to even.
We have depended a behavior of rounding to nearest.
In some case (i.e. `ActionView::Helpers::DateHelper#distance_of_time_in_words`),
we have to pass `half: :up` to `round`.

This core_ext enables to us to pass a option,
even before Ruby 2.3.

### CODE REVIEW

**Comment by yui-knk** [comment]
Maybe we should support `Rational#round` too.

**Comment by rafaelfranca** [comment]
> Maybe we should support Rational#round too (I will implement on this PR).

If it is in Ruby 2.4 yes, if not we don't need to..

**Comment by yui-knk** [comment]
I think the intention of this patch is to keep compatibility of `#round` between Ruby 2.4 and Ruby 2.3 (2.2). `Rational#round` accepts a `half` option in Ruby 2.4  https://github.com/ruby/ruby/commit/631dde2572eb78cbf09d73d23a43852ccb199bd1#diff-22b762ddc77092b556a3da28888d0f48  , so I feel it is better to add `Rational#round`.

File: activesupport/lib/active_support/core_ext/numeric/round.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# In some case (i.e. <tt>ActionView::Helpers::DateHelper#distance_of_time_in_words</tt>),
+# we have to pass <tt>half: :up</tt> to <tt>round</tt>.
+#
+# This core_ext enables to us to pass a option,
+# even before Ruby 2.3.
+if !(0.0.round(0, half: :up) rescue false)
+  require "bigdecimal"
+
+  module ActiveSupport::FloatRoundWithOption
+    def round(ndigits = 0, half: :even)
```
Comment (yui-knk) [inline activesupport/lib/active_support/core_ext/numeric/round.rb]: Is `:up` suitable for default value of `half` ?

Comment (rafaelfranca) [inline activesupport/lib/active_support/core_ext/numeric/round.rb]: For one side I don't want to change the default behavior. But for other side it seems odd to have the default different from the language. It is better to wait them to reach a decision in ruby land.

Comment (pixeltrix) [inline activesupport/lib/active_support/core_ext/numeric/round.rb]: Them changing the default behaviour from 2.3 to 2.4 seems weird - may break a bunch of finance-related Rails apps :smile:

**Comment by rafaelfranca** [comment]
Yes. This is why I opened an issue.

**Comment by rafaelfranca** [comment]
Here is the issues that I opened https://bugs.ruby-lang.org/issues/12958

**Comment by yui-knk** [comment]
After https://github.com/ruby/ruby/commit/f72be814d98f37a2f9423b46a42b35d964070d33 ruby-trunk with rails-master tests has been passed, so this PR is now useless. I will close this. rafaelfranca, thanks for your review 😄

**Pull request closed without merging**